app/src/main/java/com/example/tasky/agenda/data/remote/util/Util.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.tasky.agenda.data.remote.util
+
+import android.annotation.SuppressLint
+import com.example.tasky.agenda.data.remote.model.AgendaResponse
+import com.example.tasky.agenda.presentation.model.Agenda
+import com.example.tasky.core.data.Resource
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+fun Resource<AgendaResponse>.mapToAgenda(): Resource<List<Agenda>> {
```

Comment:
Personally I like to have a mapper for each object

Ex:
TaskMapper
EventMapper
ReminderMapper
All of this, on the "data" package, not inside the local or remote. Since you might want to convert the domain object, to a dto. Or to an entity. Or entity to dto. Etc


The problem with having it the way you currently did, is that if you want to map just a reminder (Ex: When you get a reminder by id or similar) you would need to duplicate the code

Also, the way you are doing it right now is by getting the items from API, and then converting them to a List\<Agenda\>object. What happens when you add ROOM to it? You would then need to convert this list to an entity. But what if you just want to insert a single Reminder/Task/Event? You will duplicate the code again